apps/asap-server/test/handlers/teams/index-handler.test.ts

```diff
@@ -0,0 +1,81 @@
+import { EventBridgeEvent } from 'aws-lambda';
+import {
+  indexResearchOutputByTeamHandler,
+  SquidexWebhookTeamPayload,
+} from '../../../src/handlers/teams/index-handler';
+import { TeamsEventType } from '../../../src/handlers/webhooks/webhook-teams';
+import { createEventBridgeEventMock } from '../../helpers/events';
+import {
+  algoliaClientMock,
+  algoliaIndexMock,
+} from '../../mocks/algolia-client.mock';
+import { getResearchOutputResponse } from '../../fixtures/research-output.fixtures';
+import { researchOutputControllerMock } from '../../mocks/research-outputs-controller.mock';
+
+describe('Team Research Outputs Index', () => {
+  const indexHandler = indexResearchOutputByTeamHandler(
+    researchOutputControllerMock,
+    algoliaClientMock,
+  );
+
+  afterEach(() => jest.clearAllMocks());
+
+  test('Should fetch every research output and create a record on Algolia', async () => {
+    const outputs = [
+      {
+        ...getResearchOutputResponse(),
+        id: 'research-outputs-1',
+      },
+      {
+        ...getResearchOutputResponse(),
+        id: 'research-outputs-2',
+      },
+    ];
+
+    outputs.forEach((output) =>
+      researchOutputControllerMock.fetchById.mockResolvedValueOnce(output),
+    );
+
+    const updateEvent = getEvent();
+    updateEvent.detail.payload.data.outputs.iv = outputs.map((ro) => ro.id);
+    await indexHandler(getEvent());
+
+    expect(algoliaIndexMock.saveObject).toHaveBeenCalledTimes(2);
+    expect(algoliaIndexMock.saveObject).toHaveBeenNthCalledWith(1, {
```

Comment:
we dont really care about the order of calls do we? i would simply assert that the call was made